driver-scala/src/main/scala/org/mongodb/scala/Observable.scala

```diff
@@ -464,5 +470,9 @@ trait Observable[T] extends Publisher[T] {
    * @return a single observable which emits Unit before completion.
    * @since 4.4
    */
+  @deprecated(
+    "Is no longer needed because of the `ToSingleObservableUnit` implicit class. Scheduled for removal in a major release",
+    "5.0"
+  )
   def completeWithUnit(): SingleObservable[Unit] = UnitObservable(this)
```

Comment:
Note that I propose to remove this method in the future.

Reply:
When do you propose to remove it?  Prior to 5.0, from the annotation, I gather?

Reply:
5.0 is the version from which the method is deprecated. Once it's deprecated, we'll be able to remove it in the next major release, which is 6.0.

Reply:
If we were to release a 4.x version with the deprecation, then we'll be able to delete the method in 5.0.

Reply:
So should the `"5.0"` become `"6.0"` in the annotation?

Reply:
I don't think so, because "5.0" is the value of the `since` attribute, not a version in which we promise to remove the method.

Reply:
Well that makes a lot of sense :)